sequencer/src/test/java/net/consensys/linea/jsonrpc/JsonRpcManagerTest.java

```diff
@@ -0,0 +1,264 @@
+
+    // Verify that the JSON file still exists in the directory (as the call was unsuccessful)
+    final Path rejTxRpcDir = tempDataDir.resolve("rej_tx_rpc");
+    try (Stream<Path> files = Files.list(rejTxRpcDir)) {
+      long fileCount = files.filter(path -> path.toString().endsWith(".json")).count();
+      assertThat(fileCount).as("JSON file should exist as server responded with error").isOne();
+    }
+  }
+
+  @Test
+  void firstTwoCallsErrorThenLastCallSuccessScenario() throws InterruptedException, IOException {
+    stubFor(
+        post(urlEqualTo("/"))
+            .inScenario("RPC Calls")
+            .whenScenarioStateIs(Scenario.STARTED)
+            .willReturn(aResponse().withFault(Fault.MALFORMED_RESPONSE_CHUNK))
+            .willSetStateTo("Second Call"));
+
+    stubFor(
+        post(urlEqualTo("/"))
+            .inScenario("RPC Calls")
+            .whenScenarioStateIs("Second Call")
+            .willReturn(
+                aResponse()
+                    .withStatus(500)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(
+                        "{\"jsonrpc\":\"2.0\",\"error\":{\"code\":-32000,\"message\":\"Internal error\"},\"id\":1}"))
+            .willSetStateTo("Third Call"));
+
+    stubFor(
+        post(urlEqualTo("/"))
+            .inScenario("RPC Calls")
+            .whenScenarioStateIs("Third Call")
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(
+                        "{\"jsonrpc\":\"2.0\",\"result\":{ \"status\": \"SAVED\"},\"id\":1}")));
+
+    // Prepare test data
+    final TestTransactionEvaluationContext context =
+        new TestTransactionEvaluationContext(pendingTransaction)
+            .setPendingBlockHeader(pendingBlockHeader);
+    final TransactionSelectionResult result = TransactionSelectionResult.invalid("test");
+    final Instant timestamp = Instant.now();
+
+    // Generate JSON-RPC call
+    final String jsonRpcCall =
+        JsonRpcRequestBuilder.buildRejectedTxRequest(context, result, timestamp);
+
+    // Submit the call, the scheduler will retry the failed calls
+    jsonRpcManager.submitNewJsonRpcCall(jsonRpcCall);
+
+    // Sleep to allow async processing
+    Thread.sleep(6000); // Increased sleep time to allow for three calls
+
+    // Verify that two requests were made
+    verify(exactly(3), postRequestedFor(urlEqualTo("/")).withRequestBody(equalToJson(jsonRpcCall)));
+
+    // Verify that the JSON file no longer exists in the directory (as the second call was
+    // successful)
+    Path rejTxRpcDir = tempDataDir.resolve("rej_tx_rpc");
+    try (Stream<Path> files = Files.list(rejTxRpcDir)) {
+      long fileCount = files.filter(path -> path.toString().endsWith(".json")).count();
+      assertThat(fileCount).isEqualTo(0);
+    }
+  }
+}
```

Comment:
maybe adding a test to verify that pre-existing requests are sent on start

Reply:
Done, added a new test class to cover the processing of existing json-rpc files at startup.